node-notes-cli/app.js

```diff
@@ -0,0 +1,51 @@
+import { readFile, writeFile } from 'node:fs/promises';
+
+const data = JSON.parse(await readFile('data.json'));
+const notes = data.notes;
+
+async function dataFile() {
+  const stringify = JSON.stringify(data, null, 2);
+  await writeFile('data.json', stringify);
+}
+
+async function readNotes() {
+  for (const key in notes) {
+    console.log(`${key}: ${notes[key]}`);
+  }
+}
+
+async function addNotes() {
+  const text = process.argv[3];
+  data.notes[data.nextId] = text;
+  data.nextId++;
+  dataFile();
+}
+
+async function updateNotes() {
+  const [id, newNote] = process.argv.slice(3);
+  for (const key in notes) {
+    if (key === id) {
+      notes[key] = newNote;
+    }
+  }
```

Comment:
This is much easier to do. You don't need to loop through the notes:
`notes[key] = newNote;`